### PR TITLE
#1: Implements sync policies logic from config

### LINKED ISSUE #1: Sync Policies
Right now, we have no ability to sync policies so they may match for local development. (ie. api policy, websocket policy)

We need to be able to add to the configuration file so they get created on sync.

### PR DESCRIPTION
**Pull request by erikreinert**

### CODE REVIEW

File: vault.go [hunk 1]
```diff
@@ -131,21 +142,54 @@ func SyncEngines(client *api.Client, config *Config) error {
...
+		if err != nil {
+			return err
+		}
+
+		name := FilenameWithoutExt(filename)
+
+		putErr := sys.PutPolicy(name, string(policy))
+
+		if putErr != nil {
+			return err
```
Comment (EbonJaeger) [inline vault.go]: Returning wrong error

File: file.go [hunk 2]
```diff
@@ -22,15 +28,18 @@ type ConfigSecret struct {
...
-	TargetToken   string         `json:"vault_target_token"`
+	SourceAuth         ConfigAuth     `json:"source_auth"`
+	SourceSecrets      []ConfigSecret `json:"source_secrets"`
+	SourcePoliciesPath string         `json:"source_policies_path"`
+	TargetAuth         ConfigAuth     `json:"target_auth"`
 }
 
-// GetConfig : gets configuration
+// FilenameWithoutExt : removes extension from filename
+func FilenameWithoutExt(name string) string {
```
Comment (EbonJaeger) [inline file.go:38]: One-liner only used once. Does it need to be its own function?

**Review by EbonJaeger — COMMENTED** [review]
Looks good overall, other than a couple of things.

File: file.go (hunk 2 above)
Comment (erikreinert) [reply file.go:38]: Will be used more in the future.

**Review by EbonJaeger — APPROVED** [review]

**Pull request merged**